slides/src/content/ngrx.md

```diff
@@ -0,0 +1,161 @@
+# Redux (@ngrx)
+
+---
+
+## Motivation 
+
+1. Create [Single Source of Truth](https://en.wikipedia.org/wiki/Single_source_of_truth) 
+1. [Separation of Concern](https://en.wikipedia.org/wiki/Separation_of_concerns) : Separation of Data and Logic
+1. Improve scalability by having things _react_ to events as opposed to caller dictating actions
+1. Reduce Complexity of Mental Map by promoting separation of responsibility between components
+1. Create a coherent framework so future developers will know how to add features by following convention
+1. [Uni-Directional Data Flow](http://redux.js.org/docs/basics/DataFlow.html) : Easy to reason about the interractions between actors
+
+---
+
+## Structure of Systems - Redux vs Server Architecture
+
+| Function | Angular With `@ngrx`  | Server |
+|---|---|---|
+| Data Storage | Store  |  Database  | 
+| Presentation View |  `@Component`  | Client API  |
+| Logic to massage data for Storage |  `@Effects`  | Services |
+| Communication | Action / Observables  | Function Invocation | 
+
+---
+
+## Redux as a System 
+
+<img src="content/images/redux-and-data-flow.png" width="50%"/>
+
+---
+
+## Setup Ngrx Dev-Tools
+
+FIXME
+
+---
+
+## Actions
+
+1. Responsible for Communication between parts of redux
+1. Implemented with [Observable](http://reactivex.io/rxjs/) streams
+1. Use actions instead of function Invocation
+1. Listeners are responsible to determine how to react to actions
+1. Has 2 properties, `type`, and `payload`
```

Comment:
Might want to specify that this is a convention. Redux docs have all sorts of fields which can lead to confusion.